### PR TITLE
Feature/migrate base map to leaflet

### PR DESCRIPTION
**Pull request by nlsvgtr**
# Description

Map components are based on the Leaflet library. This change migrates the code to use React Leaflet instead of the plain lib.

## Type of change

Mostly a code improvement, but presented as a feature because of size and scope.

## Documentation

Small changes will be updated after merge to master.

## Tests

Only locally

## See also

https://github.com/openstad/openstad-frontend/pull/365

### CODE REVIEW

File: src/ideas-on-map/component/ideas-on-map.jsx [hunk 1]
```diff
@@ -868,12 +868,27 @@ export default class OpenStadComponentIdeasOnMap extends OpenStadComponent {
...
+    });
+    mapConfig.defaultIcon = mapConfig.defaultIcon || { html: `<svg viewBox="0 0 26 26"><circle cx="13" cy="13" r="13" fill="#164995"/></svg>`, width: 26, height: 26, anchor: [13, 13] };
+   
     return (
 			<div id={this.divId} className={`osc-ideas-on-map osc-ideas-on-map-${this.state.status} osc-mobile-${this.state.infobarOnMobileIsOpen ? 'opened' : 'closed'}`} style={divStyle} ref={el => (this.instance = el)}>
         {filterHTML}
         {infoHTML}
         <div className={`osc-ideas-on-map-map osc-ideas-on-map-map-${this.config.display.type}`}>
-			    <Map id={this.divId + '-map'} config={{ ...this.config.map, types: this.config.types, typeField: this.config.typeField, zoomControl: this.config.display.type == 'simple' ? false : true }} ref={el => (this.map = el)}/>
+          <Suspense fallback={<div>Loading...</div>}>
```
[the remark this answers is not in the document]
Comment (nlsvgtr) [reply src/ideas-on-map/component/ideas-on-map.jsx:889]: We are currently using 16.11

File: src/base-map/css/default.less [hunk 2]
```diff
@@ -2,3 +2,25 @@
...
+  opacity: .35 !important;
+}
+
+.leaflet-div-icon {
+  border: 0 none !important;
+}
+
+// bugfix in leaflet 
+img.leaflet-marker-icon {
+  // background-image: url("./img/marker-icon.png");
```
[the remark this answers is not in the document]
Comment (nlsvgtr) [reply src/base-map/css/default.less]: Clearly not :)

File: src/image/component/image.jsx [hunk 3]
```diff
@@ -29,8 +29,8 @@ export default class OpenStadComponentImage extends OpenStadComponent {
   }
 
   componentDidMount () {
-    let elem = document.querySelector(`#${this.divId}`).parentNode;
-    this.setState({width: elem.offsetWidth})
+    let elem = document.querySelector(`#${this.divId}`) && document.querySelector(`#${this.divId}`).parentNode;
```
[the remark this answers is not in the document]
Comment (nlsvgtr) [reply src/image/component/image.jsx]: Yes, of course. Thanks.

File: src/base-map/css/default.less [hunk 4]
```diff
@@ -2,3 +2,25 @@
   width: 100%;
   height: 100%;
 }
+
+.osc-map-marker-faded {
+  opacity: .35 !important;
```
[the remark this answers is not in the document]
Comment (nlsvgtr) [reply src/base-map/css/default.less]: I seem to remember that opacity was added to the element by leaflet. But in the current version it's not, so I removed the important.

File: src/base-map/component/base-map.jsx [hunk 5]
```diff
@@ -66,12 +66,9 @@ export default class OpenStadComponentMap extends OpenStadComponent {
     let self = this;
 
     // area
-    if (self.config.area) {
-			await self.setArea(self.config.area);
-		}
+    await self.setArea(self.config.area);
```
Comment (Badmuts) [inline src/base-map/component/base-map.jsx:69]: This function is not async but its awaited?

**Review by Badmuts — APPROVED** [review]

File: src/base-map/component/base-map.jsx (hunk 5 above)
Comment (nlsvgtr) [reply src/base-map/component/base-map.jsx:69]: That does actually work, although readability wise you're right of course. Fixed, tnx.

**Pull request merged**